tests/e2e_test.go

```diff
@@ -349,3 +349,127 @@ func TestRelayTxWithWrongNonce(t *testing.T) {
 	require.Equal(t, "invalid nonce", resp1.Error.Message)
 }
+
+// Test batch request with multiple eth raw transaction
+func TestBatch_eth_sendRawTransaction(t *testing.T) {
+	resetTestServers()
+
+	var batch []*types.JsonRpcRequest
+	for i, _ := range "testing" {
+		rpcRequest := types.NewJsonRpcRequest(i, "eth_sendRawTransaction", []interface{}{testutils.TestTx_CancelAtRelay_Cancel_RawTx})
+		batch = append(batch, rpcRequest)
+	}
+	res, err := testutils.SendBatchRpcAndParseResponse(batch)
+	require.Nil(t, err, err)
+	assert.Equal(t, len(res), 7)
+}
+
+// Test batch request with different eth transaction
+func TestBatch_eth_transaction(t *testing.T) {
+	resetTestServers()
+
+	var batch []*types.JsonRpcRequest
+	req_getTransactionCount := types.NewJsonRpcRequest(1, "eth_getTransactionCount", []interface{}{testutils.TestTx_MM2_From, "latest"})
+	batch = append(batch, req_getTransactionCount)
+	// first sendRawTransaction call: rawTx that triggers the error (creates MM cache entry)
+	req_sendRawTransaction := types.NewJsonRpcRequest(1, "eth_sendRawTransaction", []interface{}{testutils.TestTx_MM2_RawTx})
+	batch = append(batch, req_sendRawTransaction)
+	// call getTxReceipt to trigger query to Tx API
+	req_getTransactionReceipt := types.NewJsonRpcRequest(1, "eth_getTransactionReceipt", []interface{}{testutils.TestTx_MM2_Hash})
+	batch = append(batch, req_getTransactionReceipt)
+
+	res, err := testutils.SendBatchRpcAndParseResponse(batch)
+	require.Nil(t, err, err)
+	assert.Equal(t, len(res), 3)
+}
+
+// Test batch request with different eth transaction
+func TestBatch_eth_call(t *testing.T) {
+	resetTestServers()
+
+	var batch []*types.JsonRpcRequest
+	// eth_call intercept
+	req := types.NewJsonRpcRequest(1, "eth_call", []interface{}{map[string]string{
+		"from": "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+		"to":   "0xf1a54b0759b58661cea17cff19dd37940a9b5f1a",
+	}})
+	batch = append(batch, req)
+	// eth_call passthrough
+	req2 := types.NewJsonRpcRequest(1, "eth_call", []interface{}{map[string]string{
+		"from": "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+		"to":   "0xf1a54b0759b58661cea17cff19dd37940a9b5f1b",
+	}})
+	batch = append(batch, req2)
+	req_getTransactionCount := types.NewJsonRpcRequest(1, "eth_getTransactionCount", []interface{}{testutils.TestTx_MM2_From, "latest"})
+	batch = append(batch, req_getTransactionCount)
+	// first sendRawTransaction call: rawTx that triggers the error (creates MM cache entry)
+	req_sendRawTransaction := types.NewJsonRpcRequest(1, "eth_sendRawTransaction", []interface{}{testutils.TestTx_MM2_RawTx})
+	batch = append(batch, req_sendRawTransaction)
+	// call getTxReceipt to trigger query to Tx API
+	req_getTransactionReceipt := types.NewJsonRpcRequest(1, "eth_getTransactionReceipt", []interface{}{testutils.TestTx_MM2_Hash})
+	batch = append(batch, req_getTransactionReceipt)
+
+	res, err := testutils.SendBatchRpcAndParseResponse(batch)
+	require.Nil(t, err, err)
+	assert.Equal(t, len(res), 5)
```

Comment:
Would be good to also check the response content

Reply:
updated